### PR TITLE
Pre-Publish: Use threshold for cover image ratio comparison.

### LINKED ISSUE #5588: Prepublish Checklist - Cover image aspect ratio check is incorrect
## Bug Description

Currently the check on the cover image aspect ration is too strict, as per discussions:

"Just as long as the cover image has a vertical based ratio it doesn’t need to be 3:4 but 3:4 is ok". 3:4 is recommended to minimize cropping across different stories but I think we can relax this a bit. @o-fernandez  WDYT?

Metadata guidance on cover image:
https://docs.google.com/presentation/d/1WVaS52rned-ZOem_gsGXkPCu48U2tR0osGbQ-P5GxQg/edit#slide=id.g83ffc12890_0_313

## Additional Context

- Plugin Version:
- WordPress Version:
- Operating System:
- Browser:

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

### PR DESCRIPTION
**Pull request by carloskelly13**
## Summary

Replace greatest common divisor method for aspect ratio with computing the scale.

## Relevant Technical Choices

Per Pascals suggestion (thank you) use a different method to test the aspect ratio!
Add a test to ensure the recommended size actually works!

## User-facing changes

Aspect ratio pre-publish checklist test for cover images 640x853 work correctly.

## Testing Instructions

1. Create a new story
2. Upload a cover image that is 640x853

Here is a sample image to use! 

star-cover

---

Fixes #5588

### CODE REVIEW

**Review by BrittanyIRL — APPROVED** [review]
Works great! Thank you for including an image to test with also :) 

I also tried adjusting the dimensions of the image in the wordPress image editor and the changes are reflected in pre publish if an existing cover image's size is altered which was cool to see.

File: assets/src/edit-story/app/prepublish/error/metadata.js [hunk 1]
```diff
@@ -127,13 +126,12 @@ export function storyCoverAspectRatio(story) {
...
-    story.featuredMedia?.height
-  );
-  const leftRatio = story.featuredMedia?.width / gcd;
-  const rightRatio = story.featuredMedia?.height / gcd;
-  if (leftRatio !== ASPECT_RATIO_LEFT || rightRatio !== ASPECT_RATIO_RIGHT) {
+  const hasCorrectAspectRatio =
+    Math.abs(
+      story.featuredMedia.width / story.featuredMedia.height -
+        ASPECT_RATIO_LEFT / ASPECT_RATIO_RIGHT
+    ) <= 0.001;
```
Comment (samwhale) [inline assets/src/edit-story/app/prepublish/error/metadata.js:133]: I haven't seen this kind of math in YEARS. I love error stuff

**Review by samwhale — APPROVED** [review]
Yay tests!

File: assets/src/edit-story/app/prepublish/error/metadata.js [hunk 2]
```diff
@@ -17,7 +17,6 @@
 /**
  * Internal dependencies
  */
-import { greatestCommonDivisor } from '../../../utils/greatestCommonDivisor';
```
Comment (swissspidy) [inline assets/src/edit-story/app/prepublish/error/metadata.js:20]: Can also delete the file and its accompanying test since it's now unused

**Review by swissspidy — APPROVED** [review]
yay :)

just one nit

**Pull request merged**